js/battle.js

```diff
@@ -1802,6 +1804,26 @@ var Side = (function () {
 		pokemon.statbarElem = this.battle.statElem.children().last();
 		this.updateStatbar(pokemon, true);
 		pokemon.side.updateSidebar();
+
+		if (pokemon.replacedIllusion && !this.battle.hackedPokemon) {
+			if (pokemon.fainted) pokemon.fainted = false; // not anymore!
+			var couldIllusion = [];
+			for (var i = 0; i < this.pokemon.length; i++) {
+				var poke = this.pokemon[i];
+				for (var i in poke.abilities) {
+					if (!poke.fainted && poke.abilities[i] === 'Illusion') couldIllusion.push(poke);
+				}
+			}
+			if (couldIllusion.length) {
+				// we know `something` had Illusion and fainted before, but not exactly what
+				// so mark the first found Illusion-legal Pokemon as fainted.
+				couldIllusion[0].hp = 0;
```

Comment:
If we're only marking the first Illusion as fainted, we don't need to collect an array, we can just collect the first candidate and `break`.